src/plugins/discover/public/services/doc_views/components/doc_viewer_table/table.tsx

```diff
@@ -57,17 +62,26 @@ export const DocViewerTable = ({
   onAddColumn,
   onRemoveColumn,
 }: DocViewRenderProps) => {
-  const showMultiFields = getServices().uiSettings.get(SHOW_MULTIFIELDS);
+  const [query, setQuery] = useState('');
+  const { storage, uiSettings } = getServices();
+  const showMultiFields = uiSettings.get(SHOW_MULTIFIELDS);
+  const storedPinnedFields = storage.get(PINNED_FIELDS_KEY);
+  const [pinnedFields, setPinnedFields] = useState<string[]>(
+    (storedPinnedFields && JSON.parse(storedPinnedFields)[indexPattern.id!]) || []
```

Comment:
In case some of the data views were removed, how we are going to clean up pinned fields list in local storage, do you have any suggestions? @kertal @majagrubic @timroes